app/models/concerns/foreman_remote_execution/host_extensions.rb

```diff
@@ -26,8 +26,9 @@ def execution_status_label(options = {})
 
     def params_with_remote_execution
       params = params_without_remote_execution
-      keys = remote_execution_ssh_keys
-      params['remote_execution_ssh_keys'] = keys unless keys.blank?
+      if (keys = remote_execution_ssh_keys).present?
+        params['remote_execution_ssh_keys'] = params['remote_execution_ssh_keys'].to_s + keys.join('\n')
```

Comment:
The user provided keys come in a string, you can't `join` them like the array, it's why I wrote the patch. user provided keys already have '\n' as global, host, or hostgroup params are strings